homeassistant/components/media_player/universal.py

```diff
@@ -0,0 +1,433 @@
+        self.hass = hass
+        self._name = name
+        self._children = children
+        self._cmds = commands
+        self._attrs = attributes
+
+        track_state_change(hass, self.dependencies, self.update_state)
+
+    def _entity_lkp(self, entity_id=None, state_attr=None, state_as_obj=True):
+        """ Looks up an entity state from hass """
+        if entity_id is None:
+            return
+
+        state_obj = self.hass.states.get(entity_id)
+
+        if state_obj is None:
+            return
+
+        if state_attr:
+            return state_obj.attributes.get(state_attr)
+        if state_as_obj:
+            return state_obj
+        return state_obj.state
+
+    def _override_or_child_attr(self, attr_name):
+        """ returns either the override or the active child for attr_name """
+        if attr_name in self._attrs:
+            return self._entity_lkp(*self._attrs[attr_name],
+                                    state_as_obj=False)
+
+        return self._child_attr(attr_name)
+
+    def _child_attr(self, attr_name):
+        """ returns the active child's attr """
+        active_child = self.active_child_state
+        return active_child.attributes.get(attr_name) if active_child else None
+
+    def _override_or_child_service(self, service_name, **service_data):
+        """ calls either a specified or active child's service """
+        if service_name in self._cmds:
+            cmd_data = self._cmds[service_name]
+            domain, service = cmd_data[CONF_SERVICE].split('.', 1)
+            self.hass.services.call(domain, service,
+                                    cmd_data[CONF_SERVICE_DATA],
+                                    blocking=True)
+            return
+
+        self._child_service(service_name, **service_data)
+
+    def _child_service(self, service_name, **service_data):
+        """ calls the active child's specified service """
+        active_child = self.active_child_state
+        service_data[ATTR_ENTITY_ID] = active_child.entity_id
+
+        self.hass.services.call(DOMAIN, service_name, service_data,
+                                blocking=True)
+
+    @property
+    def dependencies(self):
+        """ List of entity ids of entities that the mp depends on for state """
+        depend = copy(self._children)
+        for entity in self._attrs.values():
+            depend.append(entity[0])
+        return depend
+
+    @property
+    def master_state(self):
+        """ gets the master state from entity or none """
+        if CONF_STATE in self._attrs:
+            master_state = self._entity_lkp(*self._attrs[CONF_STATE],
+                                            state_as_obj=False)
+            return master_state if master_state else STATE_OFF
+        else:
+            return None
+
+    @property
+    def children(self):
+        """ Gets children and their current states """
+        return {child: self._entity_lkp(child) for child in self._children}
```

Comment:
Just use `self.hass.states.get` here, that way you can drop the `state_as_obj` parameter from `_entity_lkp`.